### PR TITLE
resolve wrong total HT price in cart if discount apply

### PR DESCRIPTION
**Pull request by InformatiqueProg**
More on issue #1476

### CODE REVIEW

File: local/modules/TheliaSmarty/Template/Plugins/DataAccessFunctions.php [hunk 1]
```diff
@@ -325,10 +325,10 @@ public function cartDataAccess($params, $smarty)
                 break;
             case "total_price":
             case "total_price_with_discount":
-                $result = $cart->getTotalAmount();
+                $result = $cart->getTotalAmount(true, $taxCountry);
```
Comment (roadster31) [inline local/modules/TheliaSmarty/Template/Plugins/DataAccessFunctions.php]: As the state is also available through Thelia\TaxEngine\TaxEngine::getDeliveryState(), you ca use it in the call to $cart->getTotalAmount()

File: local/modules/TheliaSmarty/Template/Plugins/DataAccessFunctions.php [hunk 2]
```diff
@@ -325,10 +325,10 @@ public function cartDataAccess($params, $smarty)
...
                 break;
             case "total_price_without_discount":
-                $result = $cart->getTotalAmount(false);
+                $result = $cart->getTotalAmount(false, $taxCountry);
                 break;
```
Comment (roadster31) [inline local/modules/TheliaSmarty/Template/Plugins/DataAccessFunctions.php]: ... and here too.

File: local/modules/TheliaSmarty/Template/Plugins/DataAccessFunctions.php [hunk 3]
```diff
@@ -325,10 +325,10 @@ public function cartDataAccess($params, $smarty)
...
             case "total_price":
             case "total_price_with_discount":
-                $result = $cart->getTotalAmount();
+                $result = $cart->getTotalAmount(true, $taxCountry);
                 break;
             case "total_price_without_discount":
-                $result = $cart->getTotalAmount(false);
+                $result = $cart->getTotalAmount(false, $taxCountry);
```
Comment (roadster31) [inline local/modules/TheliaSmarty/Template/Plugins/DataAccessFunctions.php]: ... and here too

File: core/lib/Thelia/Module/BaseModule.php [hunk 4]
```diff
@@ -435,7 +435,7 @@ public function getCurrentOrderTotalAmount($with_tax = true, $with_discount = tr
         /** @var Country $country */
         $country = $taxEngine->getDeliveryCountry();
 
-        $amount = $with_tax ? $cart->getTaxedAmount($country, $with_discount) : $cart->getTotalAmount($with_discount);
+        $amount = $with_tax ? $cart->getTaxedAmount($country, $with_discount) : $cart->getTotalAmount($with_discount, $country);
```
Comment (roadster31) [inline core/lib/Thelia/Module/BaseModule.php]: As the state is also available through Thelia\TaxEngine\TaxEngine::getDeliveryState(), you can use it in the call to $cart->getTotalAmount()

**Comment by roadster31** [comment]
Could you please rebase ?

**Comment by InformatiqueProg** [comment]
Done!

**Comment by roadster31** [comment]
Thanks

**Comment by InformatiqueProg** [comment]
You're welcome 😃

**Comment by roadster31** [comment]
I suspect a problem here : https://github.com/thelia/thelia/pull/2385/files#diff-0f72e814d959cda60bf282eaa9db9492R161

$cartItem is assigned by the foreach loop, and if the cart has several products, only the last on,e is considered. I'm not sure that this is working, specially if the cart item do not use the same tax (e.g., 10% and 20%).

**Comment by InformatiqueProg** [comment]
Ok, i'll check this

**Comment by roadster31** [comment]
Thanks :)

**Comment by InformatiqueProg** [comment]
You're right, an example with a product 10% VAT and another with 20% VAT
error

**Comment by roadster31** [comment]
Could you please create a new PR ?

**Comment by InformatiqueProg** [comment]
absolutely

**Comment by roadster31** [comment]
Thanks :)

**Pull request merged**